tests/python_functional/src/helpers/loggen/conftest.py

```diff
@@ -0,0 +1,187 @@
+
+        if idle_connections is not None:
+            start_parameters.append("--idle-connections={}".format(idle_connections))
+
+        if ipv6 is True:
+            start_parameters.append("--ipv6")
+
+        if debug is True:
+            start_parameters.append("--debug")
+
+        if number is not None:
+            start_parameters.append("--number={}".format(number))
+
+        if csv is True:
+            start_parameters.append("--csv")
+
+        if quiet is True:
+            start_parameters.append("--quiet")
+
+        return start_parameters
+
+    def start(
+        self, target, port, inet=None, unix=None, tcp=None, udp=None, use_ssl=None, dont_parse=None, read_file=None, skip_tokens=None, loop_reading=None,
+        rate=None, interval=None, permanent=None, syslog_proto=None, proxied=None, sdata=None, no_framing=None, active_connections=None,
+        idle_connections=None, ipv6=None, debug=None, number=None, csv=None, quiet=None,
+    ):
+
+        if self.loggen_proc is not None and self.loggen_proc.is_running():
+            raise Exception("Loggen is already running, you shouldn't call start")
+
+        instanceIndex = Loggen.get_new_instance_index()
+        self.loggen_stdout_path = Path(tc_parameters.WORKING_DIR, "loggen_stdout_{}".format(instanceIndex))
+        self.loggen_stderr_path = Path(tc_parameters.WORKING_DIR, "loggen_stderr_{}".format(instanceIndex))
+
+        self.parameters = self._decode_start_parameters(
+            inet, unix, tcp, udp, use_ssl, dont_parse, read_file, skip_tokens, loop_reading,
+            rate, interval, permanent, syslog_proto, proxied, sdata, no_framing, active_connections,
+            idle_connections, ipv6, debug, number, csv, quiet,
+        )
+
+        self.loggen_proc = ProcessExecutor().start(
+            [self.loggen_bin_path] + self.parameters + [target, port],
+            self.loggen_stdout_path,
+            self.loggen_stderr_path,
+        )
+
+        return self.loggen_proc
+
+    def stop(self):
+        if self.loggen_proc is None:
+            return
+
+        self.loggen_proc.terminate()
+        try:
+            self.loggen_proc.wait(4)
+        except TimeoutExpired:
+            self.loggen_proc.kill()
+
+        self.loggen_proc = None
+
+    def get_sent_message_count(self):
+        if not self.loggen_stderr_path.exists():
+            return 0
+
+        # loggen puts the count= messages to the stderr
+        f = open(self.loggen_stderr_path, "r")
+        content = f.read()
+        f.close()
+
+        start_pattern = "count="
+        if start_pattern not in content:
+            return 0
+        index_start = content.rindex(start_pattern) + len(start_pattern)
+
+        end_pattern = ", "
+        if end_pattern not in content:
+            return 0
+        index_end = content.find(end_pattern, index_start)
+
+        return int(content[index_start:index_end])
+
+
+@pytest.fixture
+def loggen(testcase_parameters):
```

Comment:
If the fixture is here, the testcases does not get auto-completion for `loggen`. I moved it to `src/conftest.py` and it seemed to work for me.